content/docs/digging_deeper/locks.md

```diff
@@ -0,0 +1,357 @@
+
+The AdonisJS team has created a framework-agnostic package called [Verrou](https://github.com/Julien-R44/verrou). The `@adonisjs/lock` package is based on this package, **so make sure to also read [the Verrou documentation](https://verrou.dev/docs/introduction) which is more detailed.**
+
+## Installation
+
+Install and configure the package using the following command :
+
+```sh
+node ace add @adonisjs/lock
+```
+
+:::disclosure{title="See steps performed by the add command"}
+
+1. Install the `@adonisjs/lock` package using the detected package manager.
+
+2. Registers the following service provider inside the `adonisrc.ts` file.
+    ```ts
+    {
+      providers: [
+        // ...other providers
+        () => import('@adonisjs/lock/lock_provider')
+      ]
+    }
+    ```
+
+2. Create the `config/lock.ts` file.
+
+4. Define the following environment variable alongside its validation inside the `start/env.ts` file.
+   ```ts
+   LOCK_STORE=redis
+   ```
+
+5. Optionally, create the database migration for the `locks` table if using the `database` store.
+
+:::
+
+## Configuration
+The configuration for the locks is stored inside the `config/lock.ts` file.
+
+```ts
+import env from '#start/env'
+import { defineConfig, stores } from '@adonisjs/lock'
+
+const lockConfig = defineConfig({
+  default: env.get('LOCK_STORE'),
+  stores: {
+    redis: stores.redis({}),
+
+    database: stores.database({
+      tableName: 'locks',
+    }),
+
+    memory: stores.memory()
+  },
+})
+
+export default lockConfig
+
+declare module '@adonisjs/lock/types' {
+  export interface LockStoresList extends InferLockStores<typeof lockConfig> {}
+}
+```
+
+
+<dl>
+
+<dt>
+
+default
+
+</dt>
+
+<dd>
+
+The `default` store to use for managing locks. The store is defined within the same config file under the `stores` object.
+
+</dd>
+
+<dt>
+
+stores
+
+</dt>
+
+<dd>
+
+A collection of stores you plan to use within your application. We recommend always configuring the `memory` store that could be used during testing.
+
+</dd>
+
+</dl>
+
+---
+
+
+### Environment variables
+The default lock store is defined using the `LOCK_STORE` environment variable, and therefore, you can switch between different stores in different environments. For example, use the `memory` store during testing and the `redis` store for development and production.
+
+Also, the environment variable must be validated to allow one of the pre-configured stores. The validation is defined inside the `start/env.ts` file using the `Env.schema.enum` rule.
+
+```ts
+{
+  LOCK_STORE: Env.schema.enum(['redis', 'database', 'memory'] as const),
+}
+```
+
+### Redis store
+The `redis` store has a peer dependency on the `@adonisjs/redis` package; therefore, you must configure this package before using the Redis store.
+
+Following is the list of options the Redis store accepts (alongside the shared options).
```

Comment:
`alongside the shared options`. What are the shared options? Cannot find them in Verrou docs either.

Reply:
oops. there is no shared options. this part was copy/pasted from the limiter documentation 😅